### PR TITLE
Cors2

### PR DESCRIPTION
**Pull request by jrconlin**
## Description

Set the CORS values explicitly.

## Testing

How should reviewers test?

## Issue(s)

Closes [link](link).

### CODE REVIEW

File: syncstorage/src/settings.rs [hunk 1]
```diff
@@ -239,6 +257,7 @@ impl Settings {
             "cors_allowed_methods",
             Some(vec!["DELETE", "GET", "POST", "PUT"]),
         )?;
+        s.set_default("cors_allowed_hosts", Some("*"))?;
```
Comment (pjenvey) [inline syncstorage/src/settings.rs:260]:
```suggestion
        s.set_default("cors_allowed_origin", Some("*"))?;
```

**Review by pjenvey — CHANGES_REQUESTED** [review]

**Review by pjenvey — APPROVED** [review]

**Review by pjenvey — APPROVED** [review]

File: syncstorage/src/settings.rs [hunk 2]
```diff
@@ -126,10 +126,28 @@ impl Default for Settings {
...
+            cors_allowed_methods: Some(vec![
+                "DELETE".to_owned(),
+                "GET".to_owned(),
+                "POST".to_owned(),
+                "PUT".to_owned(),
+            ]),
+            cors_allowed_headers: Some(vec![
+                "Authorization".to_owned(),
+                "Content-Type".to_owned(),
+                "UserAgent".to_owned(),
```
[the remark this answers is not in the document]
Comment (jrconlin) [reply syncstorage/src/settings.rs]: Went with the "Use the official header fields so you don't forget or misspell stuff (you idiot)" approach.

**Review by ethowitz — DISMISSED** [review]

File: syncstorage/src/settings.rs [hunk 3]
```diff
@@ -126,10 +127,28 @@ impl Default for Settings {
...
+                USER_AGENT.to_string(),
+                X_LAST_MODIFIED.to_owned(),
+                X_WEAVE_TIMESTAMP.to_owned(),
+                X_WEAVE_NEXT_OFFSET.to_owned(),
+                X_WEAVE_RECORDS.to_owned(),
+                X_WEAVE_BYTES.to_owned(),
+                X_WEAVE_TOTAL_RECORDS.to_owned(),
+                X_WEAVE_TOTAL_BYTES.to_owned(),
+                X_VERIFY_CODE.to_owned(),
+                "TEST_IDLES".to_owned(),
```
Comment (pjenvey) [inline syncstorage/src/settings.rs:149]: What's `TEST_IDLES`?

Comment (pjenvey) [reply syncstorage/src/settings.rs:149]: Whatever it is it's already part of the defaults so disregard my question.

File: syncstorage/src/settings.rs [hunk 4]
```diff
@@ -363,6 +379,15 @@ impl Settings {
         if let Some(max_age) = &self.cors_max_age {
             cors = cors.max_age(*max_age);
         }
+        // explicitly set the CORS allow origin, since Default does not
+        // appear to set the `allow-origins: *` header.
+        if let Some(origin) = &self.cors_allowed_origin {
+            if origin == "*" {
+                cors = cors.send_wildcard();
```
Comment (pjenvey) [inline syncstorage/src/settings.rs:386]: Testing this locally I see this doesn't solve the issue. Looks like we want this:
```suggestion
                cors = cors.allow_any_origin();
```

**Review by pjenvey — CHANGES_REQUESTED** [review]

**Review by pjenvey — APPROVED** [review]

**Pull request merged**